cli_tools/common/image/importer/request.go

```diff
@@ -105,6 +105,21 @@ type ImageImportRequest struct {
 	Zone                  string `name:"zone" validate:"required"`
 }
 
+// MergeBYOLIntoOSID interprets the user's --os and --byol flags, and returns a semantically-equivalent
+// tuple. The returned values follow the invariants from ImageImportRequest's validation where --byol may only
+// be specified when --os is empty (implying that detection is being used).
+//
+// For example, given `--byol --os=rhel-8`, this will return `--os=rhel-8-byol`.
+func MergeBYOLIntoOSID(osID string, byol bool) (resolvedOSID string, resolvedBYOL bool) {
+	if osID == "" || !byol {
+		return osID, byol
+	}
+	if strings.HasSuffix(osID, "byol") {
+		return osID, false
```

Comment:
What is the meaning of `resolvedBYOL`? I.e. osID in this case is byol but `false` is returned for `resolvedBYOL` which is unintuitive (not necessarily incorrect depending on how this func is used). 

More documentation and/or better return value names would be helpful here to understand what this func does and why.

Reply:
Updated the method's name, docs, and comments to clarify.

In short: The method standardizes various permutations of `--os` and `--byol` into a "standard" representation to avoid bugs in downstream code.

Reply:
I think the cause for the confusion on my side is that I didn't realize this func consumes and returns flag values. I.e. even now, `fixedBYOL`, is actuall `fixedBYOLFlagValue`. Once it's clear these are flag values, not just generic OS/BYOL variables, returning `false` for `fixedBYOL` makes sense, otherwise it looks like a bug. So, consider renaming these vars to indicate they are flag values.

Reply:
That's good feedback. Updated the arguments to be pointers and tweaked the method's names and docs.